core/src/main/java/io/kestra/core/models/tasks/runners/TaskRunner.java

```diff
@@ -101,4 +113,22 @@ public String toAbsolutePath(RunContext runContext, TaskCommands taskCommands, S
 
         return workingDir + "/" + relativePath;
     }
+    
+    /** {@inheritDoc} **/
+    @Override
+    public void kill() {
+        if (isKilled.compareAndSet(false, true)) {
```

Comment:
I don't think it's ever possible to call two times kill() on the same object as serialization will happens just after killing so the field will be set back to it's default.

Reply:
You are right, you can receive multiple kill event.

> It might be better to raise an exception when registering the kill runnable if the task has been killed in the meantime.

> In addition, I wondered what would happen if a kill event was handled by the task before a job/container was actually created

The thread will be interrupted so nothing will be created, you can test that if you"re curious ;)